### PR TITLE
feat(cargo-codspeed): build all benches in a single cargo build

### PR DESCRIPTION
**Pull request by adriencaccia**
As pointed out by @Boshen in https://github.com/oxc-project/oxc/pull/2343, we can build all the benches in a single command.

### CODE REVIEW

**Review by Boshen — APPROVED** [review]

**Review by Boshen — APPROVED** [review]

File: crates/cargo-codspeed/src/prelude.rs [hunk 1]
```diff
@@ -1,3 +1,3 @@
-pub use anyhow::{anyhow, bail, Error, Ok, Result};
+pub use anyhow::{anyhow, bail, Ok, Result};
```
Comment (art049) [inline crates/cargo-codspeed/src/prelude.rs:1]:
```suggestion
#[allow(unused_imports)]
pub use anyhow::{anyhow, bail, Error, Ok, Result};
```

File: rust-toolchain.toml [hunk 2]
```diff
@@ -1,2 +1,2 @@
 [toolchain]
-channel = "1.73.0"
+channel = "1.76.0"
```
Comment (art049) [inline rust-toolchain.toml:2]: As we discussed, it would be great to pin `clap` as well.

**Review by art049 — APPROVED** [review]

File: rust-toolchain.toml (hunk 2 above)
Comment (art049) [reply rust-toolchain.toml:2]: Maybe we could bump rust in a separate PR to make sure the performance doesn't change from the build changes

**Comment by adriencaccia** [comment]
@art049 I made the changes, let me know if it's okay to merge. I don't really know what to make of the performance improvements of the criterion sampling modes benchmarks.

**Comment by art049** [comment]
LGTM, @adriencaccia; those benches are relying on sleep. Thus we end up with syscall variance.

**Pull request merged**